hapi-base/src/main/java/ca/uhn/hl7v2/parser/FixFieldDataType.java

```diff
@@ -185,11 +199,13 @@ public static void fixMFE4(Segment segment, ModelClassFactory factory, ParserCon
      * @param dataField field number of the varies data field
      * @param defaultType default type if the typeField is empty
      * @param invalidType default type if the typeField is invalid
+     * @param shouldFix should set defaultType
      * @param factory ModelClassFactory to be used
      * @param parserConfiguration parser config
      * @throws HL7Exception if the operation fails
      */
-    public static void fix(Segment segment, int typeField, int dataField, String defaultType, String invalidType, ModelClassFactory factory, ParserConfiguration parserConfiguration)
+    public static void fix(Segment segment, int typeField, int dataField, String defaultType, String invalidType,
+            Boolean shouldFix, ModelClassFactory factory, ParserConfiguration parserConfiguration)

```

Comment:
It looks as though fix() is still fixing the field when shouldFix is false, it's just not setting the type field.
I prefer this behaviour to the blanket non-fixing of the field but in this case the property should be called SET_OBX2_PROP or something along those lines.